supertokens_python/recipe/dashboard/api/userdetails/user_put.py

```diff
@@ -0,0 +1,370 @@
+from typing import Any, Dict, Optional
+
+from supertokens_python.exceptions import raise_bad_input_exception
+from supertokens_python.recipe.dashboard.utils import (
+    get_user_for_recipe_id,
+    is_valid_recipe_id,
+)
+from supertokens_python.recipe.emailpassword import EmailPasswordRecipe
+from supertokens_python.recipe.emailpassword.asyncio import (
+    update_email_or_password as ep_update_email_or_password,
+)
+from supertokens_python.recipe.thirdpartyemailpassword.asyncio import (
+    update_email_or_password as tpep_update_email_or_password,
+)
+from supertokens_python.recipe.emailpassword.constants import FORM_FIELD_EMAIL_ID
+from supertokens_python.recipe.emailpassword.interfaces import (
+    UpdateEmailOrPasswordEmailAlreadyExistsError,
+    UpdateEmailOrPasswordUnknownUserIdError,
+)
+from supertokens_python.recipe.passwordless import PasswordlessRecipe
+from supertokens_python.recipe.passwordless.asyncio import (
+    update_user as pless_update_user,
+)
+from supertokens_python.recipe.passwordless.interfaces import (
+    UpdateUserEmailAlreadyExistsError as EmailAlreadyExistsError,
+    UpdateUserPhoneNumberAlreadyExistsError as PhoneNumberAlreadyExistsError,
+)
+from supertokens_python.recipe.passwordless.interfaces import (
+    UpdateUserUnknownUserIdError as PlessUpdateUserUnknownUserIdError,
+)
+from supertokens_python.recipe.passwordless.utils import (
+    ContactEmailOnlyConfig,
+    ContactEmailOrPhoneConfig,
+    ContactPhoneOnlyConfig,
+    default_validate_email,
+    default_validate_phone_number,
+)
+from supertokens_python.recipe.thirdpartyemailpassword import (
+    ThirdPartyEmailPasswordRecipe,
+)
+from supertokens_python.recipe.thirdpartypasswordless import (
+    ThirdPartyPasswordlessRecipe,
+)
+from supertokens_python.recipe.usermetadata import UserMetadataRecipe
+from supertokens_python.recipe.usermetadata.asyncio import update_user_metadata
+
+from ...interfaces import APIInterface, APIOptions, APIResponse
+
+
+class UserPutAPIOkResponse(APIResponse):
+    status: str = "OK"
+
+    def to_json(self) -> Dict[str, Any]:
+        return {"status": self.status}
+
+
+class UserPutAPIInvalidEmailErrorResponse(APIResponse):
+    status: str = "INVALID_EMAIL_ERROR"
+
+    def __init__(self, error: str) -> None:
+        self.error = error
+
+    def to_json(self) -> Dict[str, Any]:
+        return {"status": self.status, "error": self.error}
+
+
+class UserPutEmailAlreadyExistsAPIResponse(APIResponse):
+    status: str = "EMAIL_ALREADY_EXISTS_ERROR"
+
+    def to_json(self) -> Dict[str, Any]:
+        return {"status": self.status}
+
+
+class UserPutPhoneAlreadyExistsAPIResponse(APIResponse):
+    status: str = "PHONE_ALREADY_EXISTS_ERROR"
+
+    def to_json(self) -> Dict[str, Any]:
+        return {"status": self.status}
+
+
+class InvalidEmailError(APIResponse):
```

Comment:
Duplicate of `UserPutAPIInvalidEmailErrorResponse`

Reply:
Hmm. Removed.

Reply:
Removed.